rules/type-error.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+const tcIdentifiers = new Set([
+	'isArguments',
+	'isArray',
+	'isArrayBuffer',
+	'isArrayLike',
+	'isArrayLikeObject',
+	'isBoolean',
+	'isBuffer',
+	'isDate',
+	'isElement',
+	'isEmptyObject',
+	'isError',
+	'isFinite',
+	'isFrozen',
```

Comment:
Thanks @SamVerschueren - i really have no opinion on that. But just for completeness i could argue that testing for immutability is aiming for an constant type and throwing a `TypeError` as being the best matching error "[indicat[ing] an unsuccessful operation when none of the other NativeError objects are an appropriate indication of the failure cause](https://tc39.github.io/ecma262/#sec-native-error-types-used-in-this-standard-typeerror)".

However, since i don't bother, and you both voted for removing them, i am going to remove them. 🔨